client/src/components/molecules/MyDailyTimeRecordTable/ViewFiledChangeShiftModal/index.tsx

```diff
@@ -1,21 +1,53 @@
 import React, { FC } from 'react'
 import { Eye, X } from 'react-feather'
 
+import useUserQuery from '~/hooks/useUserQuery'
+import { Position } from '~/utils/constants/position'
 import ChangeShiftDetails from './ChangeShiftDetails'
 import ESLChangeShiftDetails from './ESLChangeShiftDetails'
 import Button from '~/components/atoms/Buttons/ButtonAction'
 import ModalTemplate from '~/components/templates/ModalTemplate'
-import { IEmployeeTimeEntry } from '~/utils/types/timeEntryTypes'
 import ModalFooter from '~/components/templates/ModalTemplate/ModalFooter'
 import ModalHeader from '~/components/templates/ModalTemplate/ModalHeader'
+import { IChangeShift, IEmployeeTimeEntry, IESLChangeShift } from '~/utils/types/timeEntryTypes'
 
 type Props = {
   isOpen: boolean
   closeModal: () => void
   timeEntry: IEmployeeTimeEntry
 }
 
-const ViewFiledChangeShiftModal: FC<Props> = ({ isOpen, closeModal, timeEntry }): JSX.Element => {
+type GetShiftDetailProps = {
+  isESLTeacher: boolean
+  changeShift: IChangeShift
+  eslChangeShift: IESLChangeShift
+}
+
+const ViewFiledChangeShiftModal: FC<Props> = (props): JSX.Element => {
+  const {
+    isOpen,
+    closeModal,
+    timeEntry: { eslChangeShift, changeShift }
+  } = props
+
+  // CURRENT USER QUERY HOOKS
+  const { handleUserQuery } = useUserQuery()
+  const { data: authUser } = handleUserQuery()
+
+  const isESLTeacher = authUser?.userById?.position.id === Position.ESL_TEACHER
+
+  const getShiftDetails = (shifts: GetShiftDetailProps): JSX.Element => {
+    const { isESLTeacher, changeShift, eslChangeShift } = shifts
+
+    if (isESLTeacher && eslChangeShift !== null) {
+      return <ESLChangeShiftDetails {...{ eslChangeShift }} />
+    } else if (!isESLTeacher && changeShift !== null) {
+      return <ChangeShiftDetails {...{ changeShift }} />
+    } else {
+      return <></>
+    }
```

Comment:
Try adjusting the code to this one. Also check if the feature is working based on the changes

```suggestion
    if (isESLTeacher && eslChangeShift !== null) {
      return <ESLChangeShiftDetails {...{ eslChangeShift }} />
    }
    if (!isESLTeacher && changeShift !== null) {
      return <ChangeShiftDetails {...{ changeShift }} />
    } 
    
    return <></>
```